### PR TITLE
 [FEATURE] excel export: formatting (resolves #526)

### LINKED ISSUE #526: Excel export: Field formatting & sum formulas
Currently all the fields in the excel export - including decimal numbers - appear to be formatted as text, which makes summing up hours difficult. (At least in LibreOffice Calc the field format has to be changed manually and then the contents have to be replaced with real numbers.)

So I suggest the following changes:

1. Add formatting to the fields according to their type (e.g. a date should have the Excel format `YYYY/mm/dd` or whatever matches config's `date_format_1`, wages the Excel format `0.00` prepended by config's `currency_sign` and so on...) This would make the decimal delimiter field in the xls export floater obsolete, as that would be locale dependant.
1. Add sum formulas for columns, where it makes sense
1. General formatting, that reduces the effort to create a version that can be printed or exported on a single PDF page:
    * emphasize headers and sums using borders and bold font
    * thicker lines between two different days
    * landscape page format (When selecting only a few columns for export, there is a chance all of them might fit on a single page)

I have already implemented those requests in [my fork](https://github.com/lentschi/kimai), since I needed a time list format like this for a customer of mine. However, if you think, that this is something the majority of your users would want, but you want some specifics changed/configurable, go ahead and tell me - I still might find some spare time. ;-)

### PR DESCRIPTION
**Pull request by lentschi**
added formatting and sum formulas to the excel export
uses `PHPExcel` (included using composer) to generate xlsx files

To try it out, you will need to run `composer install`.

@simonschaufi 

> I would prefer if the library would be included via composer but then it is not in the desired folder (core/libraries). With composer you can make sure you always get a working version. Let's figure out how we can place it there without too much trouble.

I just had to add `vendor-dir: "core/libraries"`.

It would be pretty tricky, if you'd want to have `phpunit` installed to `vendor/`, but everything else to core/libraries/. (There is https://github.com/mnsami/composer-custom-directory-installer, but that didn't work for me...)

I don't think this is necessary though. I added `bin-dir: "bin"`, so now you can run phpunit by entering `bin/phpunit`. People who just want to use kimai for production will run `composer install --no-dev` anyway.

In the long run I'd suggest making all the packages under `core/libraries` composer packages. (Than you could change the `.gitignore` file there to ignore everything but itself.) If not possible, you should maybe add an extra subpath and change composers `vendor-dir` to `core/libraries/composer`

By the way, I added composer.lock to git as it [should be in there](https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file).

### CODE REVIEW

**Comment by simonschaufi** [comment]
thank you for your investigation. btw you don't need to open another PR always. just go to your branch, make your changes, commit them and push again and the PR will be updated automatically.

File: composer.json [hunk 1]
```diff
@@ -31,5 +36,22 @@
...
+	            "type": "git",
+	            "reference": "1.8.0"
+	        },
+	        "autoload": {
+	            "psr-0": {
+	                "PHPExcel": "Classes/"
+	            }
+	        }
+	    }
+	}]
```
Comment (simonschaufi) [inline composer.json]: since this library is also on https://packagist.org/packages/phpoffice/phpexcel, i don't think we need this, or?

**Comment by lentschi** [comment]
> btw you don't need to open another PR always. just go to your branch, make your changes, commit them and push again and the PR will be updated automatically

Thanks, I know ;-) I just didn't wan't you to have thousands of commits for a single feature...

But now I pushed two new commits on the same branch:
- updated composer:
  - You were right: There is no need for the extra package source - thanks for checking!
  - I moved composer's `vendor-dir` one level down to `core/libraries/composer` (as I was pondering above), because else developers might accidentally commit packages added via composer. But of course it's up to you how you want to structure this...
- Added a small bugfix for the excel export itself

**Comment by simonschaufi** [comment]
before I can merge this, i need to talk with the team since the demo website installer also needs to be adjusted to execute a composer install after installation.

**Comment by lentschi** [comment]
Alight, thanks for the info!

**Comment by simonschaufi** [comment]
merged in develop with https://github.com/kimai/kimai/commit/57ec3a147b84d7485b9e517897e809a172e405be

**Pull request closed without merging**